qiskit/transpiler/preset_passmanagers/level1.py

```diff
@@ -252,6 +254,7 @@ def _swap_condition(property_set):
                 min_qubits=3,
                 target=target,
             ),
+            HighLevelSynthesis(),
```

Comment:
Yes, I may have missed a place or two in the previous version, thanks for spotting this. I believe it should be correct to place a call to `HighLevelSynthesis` immediately after each and every call to `UnitarySynthesis`, and I agree with @ajavadia that it makes sense to combine the two (https://github.com/Qiskit/qiskit-terra/pull/7966#discussion_r912055252). This job became much easier after the code changes for `StagedPassManager`, all of my changes only belong to `preset_passmanagers/common.py`.

Ok, contrary to what I've just written, I did not add `HighLevelSynthesis` after `UnitarySynthesis` in the "optimization loop" in `level3.py`, since `Cliffords`, etc. should have been taken care prior to this. Though, this may change if/when we want to add the ability to resynthesize and to reimplement `Cliffords`. Which way do you think it's best?